rasa/shared/nlu/training_data/message.py

```diff
@@ -179,21 +179,11 @@ def get_full_intent(self) -> Text:
             else self.get(INTENT)
         )
 
-    def get_combined_intent_response_key(self) -> Text:
-        """Get intent as it appears in training data."""
-        rasa.shared.utils.io.raise_warning(
-            "`get_combined_intent_response_key` is deprecated and "
-            "will be removed in Rasa 3.0.0. "
-            "Please use `get_full_intent` instead.",
-            category=DeprecationWarning,
-        )
-        return self.get_full_intent()
-
     @staticmethod
     def separate_intent_response_key(
         original_intent: Text,
     ) -> Tuple[Text, Optional[Text]]:
-
+        """Split intent into main intent name and optional sub-intent name."""
```

Comment:
```suggestion
        """Splits intent into main intent name and optional sub-intent name."""
```

Reply:
should we give a concrete example?

Reply:
+1 (though I see now I was removed as reviewer :) )

Reply:
@joejuzl adding you since Tobias in on vacation. So I was thinking of adding this example:
```python
        """Splits intent into main intent name and optional sub-intent name.

        For example, `"FAQ/how_to_contribute"` would be split into
        `("FAQ", "how_to_contribute")`. If there is no response delimiter, the
        second tuple item is `None`, e.g. "FAQ" would be mapped to `("FAQ", None)`.
        """
```

My only hesitation with this is that we don't actually split on `"/"` but rather the constant `RESPONSE_IDENTIFIER_DELIMITER`. So there is the risk of this docstring going stale if that delimiter ever changes.

I think this alternative while accurate is somewhat confusing:
```python
        """Splits intent into main intent name and optional sub-intent name.

        For example, `f"FAQ{RESPONSE_IDENTIFIER_DELIMITER}how_to_contribute"` would be split into
        `("FAQ", "how_to_contribute")`. If there is no response delimiter, the
        second tuple item is `None`, e.g. "FAQ" would be mapped to `("FAQ", None)`.
        """
```

Do you think the first is ok to use or do you have an alternative suggestion?

Reply:
Does it make sense to do both?
```yaml
"""Splits intent into main intent name and optional sub-intent name.

For example, `"FAQ/how_to_contribute"` would be split into
`("FAQ", "how_to_contribute")`.  The response delimiter can 
take different values (not just "\") and depends on the 
constant - RESPONSE_IDENTIFIER_DELIMITER.
If there is no response delimiter, the second tuple item 
is `None`, e.g. "FAQ" would be mapped to `("FAQ", None)`.
"""
```

Reply:
@jupyterjazz yeah I like that. Will make that change.

Reply:
Sorry just saw this. We've had similar discussion in the past about how to handle constants in docstrings - it's a tricky one :) 
The only problem I have with the proposed solution is that this is a docstring for a public method, and `RESPONSE_IDENTIFIER_DELIMITER` is not really a configurable thing, so it might be slightly confusing.

Is it not possible to fstring the value into the docstrings? That way it would always be up-to-date with the constant.